osu.Game/Users/Drawables/ClickableAvatar.cs

```diff
@@ -67,5 +63,65 @@ protected override bool OnClick(ClickEvent e)
 
             return base.OnClick(e);
         }
+
+        public partial class UserCardTooltip : VisibilityContainer, ITooltip<APIUser?>
+        {
+            public UserCardTooltip()
+            {
+                AutoSizeAxes = Axes.Both;
+            }
+
+            protected override void PopIn() => this.FadeIn(150, Easing.OutQuint);
+            protected override void PopOut() => this.Delay(150).FadeOut(500, Easing.OutQuint);
+
+            public void Move(Vector2 pos) => Position = pos;
+
+            private APIUser? user;
+
+            public void SetContent(APIUser? content)
+            {
+                if (content == user && Children.Any())
+                    return;
+
+                user = content;
+
+                if (user != null)
+                {
+                    LoadComponentAsync(new UserGridPanel(user)
+                    {
+                        Width = 300,
+                    }, panel => Child = panel);
+                }
+                else
+                {
+                    var tooltip = new OsuTooltipContainer.OsuTooltip();
+                    tooltip.SetContent(ContextMenuStrings.ViewProfile);
+                    tooltip.Show();
+
+                    Child = tooltip;
+                }
+            }
+        }
+
+        public partial class NoCardTooltip : VisibilityContainer, ITooltip<APIUser?>
+        {
+            public NoCardTooltip()
+            {
+                var tooltip = new OsuTooltipContainer.OsuTooltip();
+                tooltip.SetContent(ContextMenuStrings.ViewProfile);
+                tooltip.Show();
+
+                Child = tooltip;
+            }
+
+            protected override void PopIn() => this.FadeIn(150, Easing.OutQuint);
+            protected override void PopOut() => this.Delay(150).FadeOut(500, Easing.OutQuint);
```

Comment:
Is there a reason why this is doing easings and fades at all rather than something like

```diff
diff --git a/osu.Game/Users/Drawables/ClickableAvatar.cs b/osu.Game/Users/Drawables/ClickableAvatar.cs
index 6390acc608..ef451df95d 100644
--- a/osu.Game/Users/Drawables/ClickableAvatar.cs
+++ b/osu.Game/Users/Drawables/ClickableAvatar.cs
@@ -105,17 +105,17 @@ public void SetContent(APIUser? content)
 
         public partial class NoCardTooltip : VisibilityContainer, ITooltip<APIUser?>
         {
+            private readonly OsuTooltipContainer.OsuTooltip tooltip;
+
             public NoCardTooltip()
             {
-                var tooltip = new OsuTooltipContainer.OsuTooltip();
+                tooltip = new OsuTooltipContainer.OsuTooltip();
                 tooltip.SetContent(ContextMenuStrings.ViewProfile);
-                tooltip.Show();
-
                 Child = tooltip;
             }
 
-            protected override void PopIn() => this.FadeIn(150, Easing.OutQuint);
-            protected override void PopOut() => this.Delay(150).FadeOut(500, Easing.OutQuint);
+            protected override void PopIn() => tooltip.Show();
+            protected override void PopOut() => tooltip.Hide();
 
             public void Move(Vector2 pos) => Position = pos;
 

```

Reply:
That looks better, for sure.